src/quacc/recipes/espresso/bands.py

```diff
@@ -0,0 +1,284 @@
+"""
+This module, 'bands.py', contains recipes for performing bands and fermi surface calculations using the
+bands.x and fs.x binaries from Quantum ESPRESSO via the quacc library.
+
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from ase.dft.kpoints import bandpath
+from pymatgen.io.ase import AseAtomsAdaptor
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+
+from quacc import job
+from quacc.calculators.espresso.espresso import EspressoTemplate
+from quacc.recipes.espresso._base import base_fn
+from quacc.utils.kpts import convert_pmg_kpts
+
+if TYPE_CHECKING:
+    from typing import Any, TypedDict
+
+    from ase.atoms import Atoms
+
+    from quacc.schemas._aliases.ase import RunSchema
+
+    class BandsSchema(TypedDict):
+        bands: RunSchema
+        bands_pp: RunSchema
+        fermi_surface: RunSchema
+
+
+@job
+def bands_job(
+    atoms: Atoms,
+    prev_dir: str | Path,
+    run_bands_pp: bool = True,
+    run_fermi_surface: bool = False,
+    primitive: bool = True,
+    parallel_info: dict[str] | None = None,
+    test_run: bool = False,
+    job_params: dict[str, Any] | None = None,
+) -> BandsSchema:
+    """
+    Function to compute bands structure and fermi surface using pw.x, bands.x and fs.x.
+    This is all done in a single job as a multi-step process.
+
+    1. A pw.x non-self consistent calculation
+        - name: "bands"
```

Comment:
Please also include the hyperlinks for these, e.g. `[quacc.recipes.espresso.bands.bands_pp][]`